packages/api-pool-updater/updatePools.js

```diff
@@ -11,18 +11,32 @@ import { getPoolsKey } from '../../utils/getPoolsKey'
  * @returns
  */
 export const updatePools = async (event, chainId) => {
-  const pools = await getPools(chainId, fetch)
+  const updatedPools = await getPools(chainId, fetch)
 
-  if (!pools || pools.length !== poolCountForChain(chainId)) {
+  if (!updatedPools) {
     event.waitUntil(log(new Error('No pools fetched during update'), event.request))
+    return false
   }
 
-  event.waitUntil(POOLS.put(getPoolsKey(chainId), JSON.stringify(pools)), {
+  const storedPools = JSON.parse(await POOLS.get(getPoolsKey(chainId)))
+
+  updatedPools.map((updatedPool) => {
```

Comment:
instead of `updatedPool` this could simply be `pool`. Or if you prefer, `poolToUpdate`